server/repos/reposController.js

```diff
@@ -0,0 +1,64 @@
+var url = require('url');
+var utils = require('./reposUtils');
+var Promise = require('bluebird')
+var request = require('request');
+
+module.exports = {
+  getRepo: function(req, res) {
+    var href = url.parse(req.url);
```

Comment:
use express routes instead of manually parsing url imo
app.get('/v1/users/:username', function(request, response, next) {
  var username = request.params.username;
  findUserByUsername(username, function(error, user) {
    if (error) return next(error);
    return response.render('user', user);
  });
});